pandas/tests/io/parser/test_common.py

```diff
@@ -899,8 +899,12 @@ def test_nonexistent_path(all_parsers):
 
     msg = ("does not exist" if parser.engine == "c"
            else r"\[Errno 2\]")
-    with pytest.raises(compat.FileNotFoundError, match=msg):
+    with pytest.raises(compat.FileNotFoundError, match=msg) as e:
         parser.read_csv(path)
+    try:
```

Comment:
why would this be needed?

Reply:
It seems that depending on the parser used (and the test is run with 3 different parsers) the `filename` parameter of the exception can either be a `bytes` or a `str`. This was my attempt to account for this.

Is there a better solution for this or a way to avoid it in the first place?